src/internal/operations/backup.go

```diff
@@ -117,9 +117,65 @@ func (op BackupOperation) validate() error {
 // pointer wrapping the values, while those values
 // get populated asynchronously.
 type backupStats struct {
-	k             *kopia.BackupStats
-	ctrl          *data.CollectionStats
-	resourceCount int
+	k                   *kopia.BackupStats
+	ctrl                *data.CollectionStats
+	resourceCount       int
+	hasNewDetailEntries bool
+}
+
+// To qualify as an assist backup, all of the following must be true:
+// 1. new detail entries were produced during this operation
+// 2. The updated details file was persisted without error
+// 3. we have a valid snapshot ID
+// 4. we don't have any non-recoverable errors
+// 5. we recorded recoverable errors
+// 6. We are not running in best effort mode. Reason being that there is
+// no way to distinguish assist backups from merge backups in best effort mode
+// Primary reason for persisting assist backup models is to ensure we don't
+// lose corso extension data(deets) for items which were downloaded and
+// processed by kopia during this backup operation.
+// Note: kopia.DetailsMergeInfoer doesn't impact decision making for creating
+// assist backups. It may be empty if it’s the very first backup so there is no
+// merge base to source base details from, or non-empty, if there was a merge base.
+// In summary, if there are no new deets, no new extension data was produced
+// and hence no need to persist assist backup model.
+func isAssistBackup(
+	newDeetsProduced bool,
+	snapID, ssid string,
+	failurePolicy control.FailurePolicy,
+	err *fault.Bus,
+) bool {
+	return newDeetsProduced &&
+		snapID != "" &&
+		ssid != "" &&
+		err.Failure() == nil &&
+		len(err.Recovered()) > 0 &&
+		failurePolicy != control.BestEffort
```

Comment:
if the best effort mode check is removed from isMergeBackup it should be removed from here too